src/main/java/org/openstreetmap/atlas/utilities/arrays/LongArray.java

```diff
@@ -53,6 +58,12 @@ public LongArray(final long maximumSize, final int memoryBlockSize, final int su
         super(maximumSize, memoryBlockSize, subArraySize);
     }
 
+    @Override
+    public ProtoAdapter getProtoAdapter()
+    {
+        return this.protoLongArrayAdapter;
```

Comment:
How about returning a new one every time? That would let you not have any transient variable. Less risky if you try to de-serialize a Java based Atlas and then re-save it... You would get a NPE here, no?